cypress/integration/unit_tests/checkbox_test.js

```diff
@@ -92,35 +101,238 @@ describe('Unit test for toggleLayerOn', () => {
+    expect(map.overlayMapTypes).to.have.length(0);
+    expect(map.overlayMapTypes.getAt(3)).to.be.undefined;
+
+    // Before EE rendering finishes, toggle layer off.
+    toggleLayerOff(3, map);
+    // Overlay list now has null instead of undefined, but no biggie.
+    expect(map.overlayMapTypes.getAt(3)).to.be.null;
+
+    // Still before evaluation finishes, toggle back on.
+    const togglePromise = toggleLayerOn(mockFirebaseLayers[3], map);
+    expect(togglePromise).equals(promise);
+    // Still null.
+    expect(map.overlayMapTypes.getAt(3)).to.be.null;
+
+    // Loading can't finish until EE evaluation finishes, which we've frozen.
+    expect(loadingFinishedStub).to.not.be.called;
+
+    // Release evaluation.
+    latch.release();
+    cy.wrap(promise).then(() => {
+      expect(loadingFinishedStub).to.be.calledOnce;
+      expect(map.overlayMapTypes.getAt(3)).to.not.be.null;
+    });
+  });
+
+  it('tests score layer and deck management', () => {
+    // Re-initialize deck, this time with a spy so we can observe what happens.
+    const deckSpy = cy.spy(deck, 'GoogleMapsOverlay');
+    setMapToDrawLayersOn(null);
+    const overlaySpy = cy.spy(deckSpy.returnValues[0], 'setProps');
+    let numCalls = 0;
+    /** @return {array<deck.GeoJsonLayer>} */
+    function getLatestLayer() {
+      return overlaySpy.args[numCalls++][0].layers;
+    }
+
+    let resolveFunction = null;
+    const promise = new Promise((resolve) => resolveFunction = resolve);
+
+    const scorePromise = addScoreLayer(promise);
+    expect(scorePromise).to.not.be.null;
+    // Nothing happens until promise we passed in is resolved.
+    expect(overlaySpy).to.not.be.called;
+    const promiseResult = ['a'];
+    const secondPromiseResult = ['b'];
+    resolveFunction(promiseResult);
+
+    cy.wrap(scorePromise)
+        .then(() => {
+          expect(overlaySpy).to.be.calledOnce;
+          const firstLayers = getLatestLayer();
+          expect(firstLayers).to.have.length(3);
+          const props = firstLayers[2].props;
+          expect(props.data).to.eql(promiseResult);
+          expect(props.visible).to.be.true;
+          expect(props.id).to.eql(scoreLayerName);
+
+          // Test remove.
+          removeScoreLayer();
+          expect(overlaySpy).to.be.calledTwice;
+          const secondLayers = getLatestLayer();
+          expect(secondLayers).to.have.length(2);
+          expect(secondLayers[1].props.id).to.not.eql(scoreLayerName);
+
+          // Add score layer back in, with a different value.
+          const secondPromise =
+              new Promise((resolve) => resolveFunction = resolve);
+          const secondScorePromise = addScoreLayer(secondPromise);
+          // Nothing happens until promise is resolved.
+          expect(overlaySpy).to.be.calledTwice;
+          resolveFunction(secondPromiseResult);
+          return secondScorePromise;
+        })
+        .then(() => {
+          expect(overlaySpy).to.be.calledThrice;
```

Comment:
Is this cumulative?

Reply:
Yup. Sinon stubs appear to have a resetHistory method, but sadly Cypress's wrapper doesn't appear to expose that.